src/content/lesson/everything-you-need-to-start-using-sqlalchemy.md

```diff
@@ -164,15 +164,15 @@ But we have a list of extra ingredients we can add to it: first we choose meat b
 ```jsx
 INSERT INTO class PIZZA(4, 'meat');
 
-COMMIT; 
+session.commit(); 
```

Comment:
Something like this:
```py

# we insert a new ingredient into out pizza
ingredient = Ingredient()
ingredient.name = 'meat'
ingredient.id = 4
db.session.add(ingredient)

# now we COMMIT the transaction and save it into the database
db.session.commit()

ingredient = Ingredient.query.get(4)
ingredient.name = mushrooms

# save a checkpoint
checkpoint_a = db.session.begin_nested()

# add pepperoni
ingredient = Ingredient()
ingredient.name = 'pepperoni'
db.session.add(ingredient)

# one last checkpoint before adding bacon ingredient
checkpoint_b = db.session.begin_nested()

# add bacon
ingredient = Ingredient()
ingredient.name = 'bacon'
db.session.add(ingredient)
```

Now our 'PIZZA' has the following ingredients:

![SQL](../../assets/images/sql-3.png)

Let's say we have decided we no longer want bacon, so we use ROLLBACK to checkpoint B:

```jsx
checkpoint_b.rollback()
```
and our pizza looks like this:

![SQL](../../assets/images/sql-4.png)

....I'm a bit hungry after reading this lesson!! aren't you??